floris/tools/uncertainty_interface.py

```diff
@@ -1,337 +1,103 @@
-            # value (e.g., 0 deg) when evaluating uncertainty. Evaluating
-            # wind direction uncertainty like this would essentially come
-            # down to a Gaussian smoothing of FLORIS solutions over the
-            # wind directions. This can also be really fast, since it would
-            # not require any additional calculations compared to the
-            # non-uncertainty FLORIS evaluation.
-            yaw_angles_probablistic = np.vstack([
-                np.expand_dims(yaw_angles_nominal, axis=0)
-                for _ in unc_pmfs["wd_unc"]
-            ])
-        else:
-            # Fix yaw angles in the absolute (compass) reference frame,
-            # meaning that for each probablistic wind direction evaluation,
-            # our probablistic (relative) yaw angle evaluated goes into
-            # the opposite direction. For example, a probablistic wind
-            # direction 3 deg above the nominal value means that we evaluate
-            # it with a relative yaw angle that is 3 deg below its nominal
-            # value.
-            yaw_angles_probablistic = np.vstack([
-                np.expand_dims(yaw_angles_nominal, axis=0) - dy
-                for dy in unc_pmfs["wd_unc"]
-            ])
-
-        self.wd_array_probablistic = wd_array_probablistic
-        self.yaw_angles_probablistic = yaw_angles_probablistic
-
-    def _reassign_yaw_angles(self, yaw_angles=None):
-        # Overwrite the yaw angles in the FlorisInterface object
-        if yaw_angles is not None:
-            self.fi.floris.farm.yaw_angles = yaw_angles
-
-    # Public methods
-
-    def copy(self):
-        """Create an independent copy of the current UncertaintyInterface
-        object"""
-        fi_unc_copy = copy.deepcopy(self)
-        fi_unc_copy.fi = self.fi.copy()
-        return fi_unc_copy
-
-    def reinitialize_uncertainty(
+        wd_resolution (float, optional): The resolution of wind direction, in degrees.
+            Defaults to 1.0.
+        ws_resolution (float, optional): The resolution of wind speed, in m/s. Defaults to 1.0.
+        ti_resolution (float, optional): The resolution of turbulence intensity. Defaults to 0.01.
+        yaw_resolution (float, optional): The resolution of yaw angle, in degrees. Defaults to 1.0.
+        power_setpoint_resolution (int, optional): The resolution of power setpoints, in kW.
+            Defaults to 100.
+        wd_std (float, optional): The standard deviation of wind direction. Defaults to 3.0.
+        wd_sample_points (list[float], optional): The sample points for wind direction.
+            If not provided, defaults to [-2 * wd_std, -1 * wd_std, 0, wd_std, 2 * wd_std].
+        verbose (bool, optional): Verbosity flag for printing messages. Defaults to False.
+
+    """
+    def __init__(
         self,
-        unc_options=None,
-        unc_pmfs=None,
-        fix_yaw_in_relative_frame=None
+        configuration: dict | str | Path,
```

Comment:
Are `str` and `Path` supported?

Reply:
I'm not sure, I think I copied this from the type hint in FlorisInterface, and this is just getting passed downwards to there